src/Tribe/Assets.php

```diff
@@ -0,0 +1,367 @@
+		$defaults = array(
+			'action'    => null,
+			'priority'  => 10,
+			'file'      => false,
+			'type'      => null,
+			'deps'      => array(),
+			'version'   => $version,
+			'media'     => 'all',
+			'in_footer' => true,
+			'localize'  => array(),
+			'filter'    => array(),
+		);
+
+		// Merge Arguments
+		$asset = (object) wp_parse_args( $arguments, $defaults );
+
+		// Enforce these one
+		$asset->slug        = $slug;
+		$asset->file        = $file;
+		$asset->deps        = $deps;
+		$asset->origin      = $origin;
+		$asset->origin_name = $origin_name;
+		$asset->action      = $action;
+
+		// If we don't have a type on the arguments we grab from the File path
+		if ( is_null( $asset->type ) ) {
+			if ( substr( $asset->file, -3, 3 ) === '.js' ) {
+				$asset->type = 'js';
+			} elseif ( substr( $asset->file, -4, 4 ) === '.css' ) {
+				$asset->type = 'css';
+			}
+		}
+
+		// If asset type is wrong don't register
+		if ( ! in_array( $asset->type, array( 'js', 'css' ) ) ) {
+			return false;
+		}
+
+		/**
+		 * Deprecated filter to allow changing version based on the type of Asset
+		 *
+		 * @todo remove on 4.6
+		 * @deprecated 4.3
+		 *
+		 * @param string $version
+		 */
+		$asset->version = apply_filters( "tribe_events_{$asset->type}_version", $asset->version );
+
+		/**
+		 * Filter to change version number on assets
+		 *
+		 * @param string $version
+		 * @param object $asset
+		 */
+		$asset->version = apply_filters( 'tribe_asset_version', $asset->version, $asset );
+
+		// Clean these
+		$asset->priority  = absint( $asset->priority );
+		$asset->in_footer = (bool) $asset->in_footer;
+		$asset->media     = esc_attr( $asset->media );
+
+		// Ensures that we have a priority over 1
+		if ( $asset->priority < 1 ) {
+			$asset->priority = 1;
+		}
+
+		$is_vendor = strpos( $asset->file, 'vendor/' ) !== false ? true : false;
+
+		// Setup the actual URL
+		$asset->url = $this->maybe_get_min_file( tribe_resource_url( $asset->file, false, ( $is_vendor ? '' : null ), $asset->origin ) );
+
+		// If you are passing localize, you need `name` and `data`
+		if ( ! empty( $asset->localize ) && ( is_array( $asset->localize ) || is_object( $asset->localize ) ) ) {
+			$asset->localize = (object) $asset->localize;
+
+			// if we don't have both reset localize
+			if ( ! isset( $asset->localize->data, $asset->localize->name ) ) {
+				$asset->localize = array();
+			}
+		}
+
+		/**
+		 * Filter an Asset loading variables
+		 *
+		 * @param object $asset
+		 */
+		$asset = apply_filters( 'tribe_asset', $asset );
+
+		// If we have any type of input on Filter we hook it
+		if ( ! empty( $asset->filter ) ) {
+			// This allows us to
+			if ( is_array( $asset->filter ) && is_callable( $asset->filter ) ) {
+				$asset->filter = array( $asset->filter );
+			}
+
+			// To make this work we will use the 0 as the number of arguments
+			foreach ( $asset->filter as $filter ) {
+				add_filter( 'tribe_asset_enqueue_' . $asset->slug, $filter, 10, 0 );
```

Comment:
Very true, I would love to hope on call to discuss this problem further, hard to solve this problem.